app/services/medium_providers/provider.rb

```diff
@@ -41,9 +41,11 @@ def additional_media
       parse_media(media) || []
     end
 
-    # Returns unique string representing current installation medium.
-    def unique_id
-      @unique_id ||= Base64.urlsafe_encode64(Digest::SHA1.digest(medium_uri.to_s), padding: false)
+    # Returns unique string representing current installation medium. PXE prefix path is passed as a parameter
+    # and it is important to take it into account when generating unique string as some operating systems (e.g.
+    # Debian/Ubuntu) has a constant base URL for all versions.
+    def unique_id(path = '')
+      raise "unique_id is not implemented for #{self.class.name}"
     end
```

Comment:
I am making the method abstract, this should really not be here. Each implementation should do it on its own.

I wonder, since now it's only in the default implementation, I can probably access the pxedir via operatingsystem and remove the parameter! Let me look.